### DISCUSSION TITLE
Sorting entities in views/groups

### DISCUSSION THREAD

**Discussion by menganha** [skypjack/entt#1155] [Q&A] [answered]
Hello everybody,

I'm sorting the rendered entities (the order defined, e.g., by a component that specifies how far we are from the camera) and I was wondering what's the approach that works best for you. 

My current approach is to sort at each frame by using the sort method of the registry
[https://github.com/skypjack/entt/wiki/Crash-Course:-entity-component-system#sorting-is-it-possible](https://github.com/skypjack/entt/wiki/Crash-Course:-entity-component-system#sorting-is-it-possible)
but I'm worried this creates unnecessary overhead since in my specific use-case I'm not creating nor destroying entities that often, i.e., I probably don't need to resort them at every frame.

Would it be better to just sort the components *on demand* ? Meaning to call the sort function on construction or on destruction of the relevant component, i.e., using the *on_...* methods to observe changes of the registry. Would this approach cost the same or less as the per-frame resorting?
[https://github.com/skypjack/entt/wiki/Crash-Course:-entity-component-system#observe-changes](https://github.com/skypjack/entt/wiki/Crash-Course:-entity-component-system#observe-changes)

Thanks in advance,
Menganha

**Comment by skypjack** [comment, marked as answer]
If you added N elements of the same type in a tick, the last method would sort N times a pool vs a single time of the original approach.
If you really want to lazy sort things on demand, use signals or another approach to raise a flag and check this flag before sorting.
This way you know that you're only sorting once and when needed. This is probably the best of the two worlds.

**Reply by menganha** [reply]
I see. I hadn't tought about the consequences of that particular case (repeated sorting callbacks from adding more than one element on the same tick). I think for the scope of the application I'm working, the one sorting per tick should be enough. Later I can go for the lazy sort :) if really needed.
Many thanks
